apps/heft/src/plugins/TypescriptPlugin/TypescriptBuilder.ts

```diff
@@ -0,0 +1,1029 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import {
+  Colors,
+  JsonFile,
+  FileSystemStats,
+  IFileSystemCreateLinkOptions,
+  IColorableSequence
+} from '@rushstack/node-core-library';
+import * as crypto from 'crypto';
+import { Tslint, Typescript } from '@microsoft/rush-stack-compiler-3.7';
+import {
+  ExtendedTypescript,
+  IEmitResolver,
+  IExtendedEmitResult,
+  IEmitHost,
+  IEmitTransformers,
+  IExtendedProgram,
+  IExtendedSourceFile,
+  IResolveModuleNameResolutionHost
+} from './privateTypings/ExtendedTypescript';
+
+import { SubprocessRunnerBase } from '../../utilities/subprocess/SubprocessRunnerBase';
+import { Async } from '../../utilities/Async';
+import { resolvePackagePath } from '../../utilities/resolvePackage';
+import { IExtendedLinter } from './privateTypings/ExtendedTslint';
+import { IEmitModuleKindBase, ISharedTypescriptConfiguration } from '../../cli/actions/BuildAction';
+
+const ASYNC_LIMIT: number = 100;
```

Comment:
This should be configurable. A default of `100` seems very high:
- A hard drive cannot efficiently write 100 files simultaneously
- In a Rush repo with `--parallelism` equal to the number of CPU cores, the task switching overhead would be pretty inefficient for 100 concurrent operations
- On MacOS the  maximum number of file handles can be relatively low, see https://github.com/microsoft/rushstack/issues/1934#issuecomment-648489001